bin/fin

```diff
@@ -6287,6 +6344,631 @@ set_project_root_permissions ()
+		provider_pull_files
+	fi
+
+	echo-green "Pull Completed"
+}
+
+# Provider Pull Init Subcommand
+# Used specifically for pulling the repo from a provider.
+provider_pull_init ()
+{
+	echo-green "Starting Pull Init Process."
+	local project_directory=${ARGV[1]:-${HOSTING_SITE}}
+
+	# Check to see if project_directory already exists.
+	if [[ -d ${project_directory} ]]; then
+		echo-error "Project Directory ${project_directory} already exists."
+		exit 1
+	fi
+
+	provider_pull_find_function ${HOSTING_PLATFORM} 'init' 1 "Init function not currently supported for ${HOSTING_PLATFORM}."
+	echo-green "Pulling code complete."
+
+	# Change into the project directory.
+	cd "${project_directory}"
+
+	# Check to see if .docksal directory exists.
+	if [[ ! -d .docksal ]]; then
+		_confirm "Project is not a docksal project. Would you like to generate config?"
+		config_generate
+	fi
+
+	# Set the Current Hosting Platform and Hosting Site for project.
+	config_set "HOSTING_PLATFORM=${HOSTING_PLATFORM}" > /dev/null
+	config_set "HOSTING_SITE=${HOSTING_SITE}" > /dev/null
+	config_set "HOSTING_ENV=${HOSTING_ENV}" > /dev/null
+	addon_file_path="$(addon_get_path init)"
+	if [[ ! -z "$addon_file_path" ]] && [[ -x "$addon_file_path" ]]; then
+		if _confirm "Init command found for project run?" --no-exit; then
+			addon_execute "$addon_file_path"
+		fi
+	fi
+}
+
+# Generic Precheck function to check if HOSTING_ENV variable is set.
+provider_pull_precheck ()
+{
+	# After Init Process Let's Check to Make sure HOSTING_ENV has been set.
+	if [[ "${HOSTING_ENV}" == "" ]]; then
+		echo-error "HOSTING_ENV variable is required to continue."
+		exit 1
+	fi
+}
+
+# Provider Pull Code Subcommand
+# Used specifically for pulling the repo from a provider.
+provider_pull_code ()
+{
+	echo-green "Pulling code"
+	provider_pull_find_function ${HOSTING_PLATFORM} 'code' 1
+	echo-green "Code Pull Successful"
+}
+
+# Provider Pull DB Subcommand
+# Used specifically for pulling the db from a provider.
+provider_pull_db ()
+{
+	_confirm "Existing database will be truncated, Continue?"
+	echo-green "Pulling DB"
+
+	# Variables to set for local use.
+	local __db_user="${DBUSER:-root}"
+	local __db_password="${DBPASS:-root}"
+	local __db_database="${DBNAME:-default}"
+
+	# Set the default db_name just incase we need to cache this for Acquia or other providers.
+	local db_name=${ARGV[1]:-"default"}
+	# Set the db file that will be stored within the cli container
+	local db_file="/tmp/${db_name}.${HOSTING_ENV}.${HOSTING_SITE}.sql.gz"
+	# Check to see if the file exists in the cli container
+	local file_exists=$(_exec "ls -1 /tmp/**  2> /dev/null | grep \"${db_file}\"\$ | wc -l | tr -d '[:space:]'")
+	# Get the time of that file within the cli container.
+	local file_time=$(_exec "find ${db_file} -mmin +360 2> /dev/null | tr -d '[:space:]'")
+	# Get the time from 24 hours ago.
+	local yesterday=$(_exec date --date='yesterday' "+%Y-%m-%d %H:%M:%S");
```

Comment:
Wouldn't it be more robust to compare dates in unix timestamp format? `date --date='yesterday' +%s`